### PR TITLE
docs(sns): remove an unused variable declaration in Archive Policy

### LINKED ISSUE #29590: docs(sns): remove an unused variable declaration in Archive Policy
### Describe the issue

Archive Policy parameter in the aws_cdk_sns document has an unused variable declaration.

Currently the documentation says:
***
Example with an archive policy for SQS:
```ts
declare const role: iam.Role;
const topic = new sns.Topic(this, 'MyTopic', {
   fifo: true,
   messageRetentionPeriodInDays: 7,
});
```
***

I think the following is better.
***
Example with an archive policy:
```ts
const topic = new sns.Topic(this, 'MyTopic', {
   fifo: true,
   messageRetentionPeriodInDays: 7,
});
```
***
Removed "for SQS" and "declare const role: iam.Role;".

### Links

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_sns-readme.html#archive-policy

### PR DESCRIPTION
**Pull request by jun1-t**
Closes #29590

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍
Btw, it looks like an unused variable declaration, not a typo.
If you want to adjust the title accordingly.

**Comment by jun1-t** [comment]
[lpizzinidev](https://github.com/lpizzinidev) Thank you for your review. Indeed, it's not a typo, so I have corrected the title.

File: packages/aws-cdk-lib/aws-sns/README.md [hunk 1]
```diff
@@ -294,10 +294,9 @@ Message archiving provides the ability to archive a single copy of all messages
 You can store published messages within your topic by enabling the message archive policy on the topic, which enables message archiving for all subscriptions linked to that topic.
 Messages can be archived for a minimum of one day to a maximum of 365 days.
 
-Example with a archive policy for SQS:
+Example with a archive policy:
```
Comment (scanlonp) [inline packages/aws-cdk-lib/aws-sns/README.md:297]:
```suggestion
Example with an archive policy:
```

**Review by scanlonp — APPROVED** [review]
Looks good. Thanks for handling this!

**Pull request merged**